### ISSUE TITLE
.NET Standard 2.1 support

### ISSUE THREAD

**Issue by hez2010** [Tornhoof/SpanJson#111]
Please add .NET Standard 2.1 support, thanks.

**Comment by Tornhoof** [comment]
Thank you for raising this issue.
I don't think that .NET Standard 2.1 will work here, the current version only works with .NET Core 2.1+.
Assuming that another Framework, e.g. Mono adds Support for .NET Standard 2.1, it is highly unlikely that the current code base will work with Mono on e.g. ARM platforms.

The next major Version of SpanJson will most likely target .NET Core 3.0 to get support for intrinsics etc.

**Comment by hez2010** [comment]
Thanks for your reply. I'll close the issue.

**Comment by silkfire** [comment]
@Tornhoof Would you reconsider adding .NET Standard 2.1 support? API methods like `RuntimeHelpers.IsReferenceOrContainsReferences<T>` can be found there so shouldn't be an issue.

**Comment by Tornhoof** [comment]
Has anything changed w.r.t .NET Standard 2.1 I'm not aware of?
Fwiw, future development of SpanJson will probably go more the route of source generators, as they are more or less the only real way to write proper async serializers. The current Expression Tree concept is a dead end, due to MS more or less abandoning expression trees for code gen. I expect the next major version of SpanJson to be async only via pipelines and source generators. But that will only happen if I get the performance even remotely into a similar ballpark as the current version is.

**Comment by silkfire** [comment]
Sounds very interesting, I'm certain SGs would be of great benefit for a library like this.

**Comment by silkfire** [comment]
Hello @Tornhoof , have you considered revisiting the idea for SpanJson to utilize the power of source generators and pipelines? Microsoft is doing something similar with the `System.Text.Json` API in .NET 6: https://devblogs.microsoft.com/dotnet/announcing-net-6-preview-5/

**Comment by Tornhoof** [comment]
Yeah I did, the bottomline is, that all my attempts to write proper async serializers with the current SpanJson approach of having ref structs failed, as the async state machines can't be used with ref structs (the reader and writer are ref structs), it's possible to work around for some types (basic pocos), but things like readonly types with constructor assignments (like records) or even lists do not work nicely.
As for pipelines, most of my attempts quit were a bit slower (15%+) than the basic arraypool buffer now, but I expect that can be solved, but after not finding a good solution for the async problem I stopped here fairly early.

MS is bypassing the whole async problem, as their converters remains sync and they handle async (for e.g. flushing) outside of their converters.
